### PR TITLE
GH-48: Reinvent ip_to_cron.rb from foreman

### PR DESCRIPTION
**Pull request by jyaworski**

### CODE REVIEW

**Comment by jyaworski** [comment]
This needs signoff from the original authors of ip_to_cron.rb due to GPL -> Apache.

https://github.com/theforeman/puppet-puppet/blob/master/lib/puppet/parser/functions/ip_to_cron.rb

**Comment by roidelapluie** [comment]
Apparently the authors are @mcanevet @gregSutcliffe

**Comment by roidelapluie** [comment]
@mcanevet @gregsutcliffe we need your agreement to extract the ip_to_cron function to this lib, because the foreman-puppet module is GPLv3 while this module is apache.

**Comment by mcanevet** [comment]
@roidelapluie it's OK for me

File: lib/puppet/parser/functions/ip_to_cron.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
+#
+# ip_to_cron.rb
+#
+# Copyright 2016 Vox Pupuli
```
Comment (roidelapluie) [inline lib/puppet/parser/functions/ip_to_cron.rb]: missing original authors copyright

**Comment by GregSutcliffe** [comment]
@roidelapluie I'm OK with that, go ahead. Sorry for the delay!

File: lib/puppet/parser/functions/ip_to_cron.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+  notice ip_to_cron(2500)
+  notice ip_to_cron(5500)
+  notice ip_to_cron(7500)
+
+The results will be as follows, varying on FQDN and network addresses:
+
+  notice: Scope(Class[main]):
+  notice: Scope(Class[main]):
+  notice: Scope(Class[main]):
```
Comment (hunner) [inline lib/puppet/parser/functions/ip_to_cron.rb]: Why is the return value empty? What does this illustrate?

File: spec/functions/ip_to_cron_spec.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+  it 'should return an array of length 2 with runinterval <= 3600' do
+    expect(subject.call([3500])).to be_an Array
+    expect(subject.call([3500]).length).to eq(2)
+  end
+
+  it 'should return an array of length 2 with runinterval > 3600' do
+    expect(subject.call([7200])).to be_an Array
+    expect(subject.call([7200]).length).to eq(2)
+  end
```
Comment (hunner) [inline spec/functions/ip_to_cron_spec.rb]: Shouldn't this test the actual return value? Just stub the fqdn and ipaddress facts so it's consistent.

File: lib/puppet/parser/functions/ip_to_cron.rb [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+module Puppet::Parser::Functions
+  newfunction(:ip_to_cron, :type => :rvalue, :doc => <<-EOS
+Returns an array of valid cron times given an optional runinterval
```
Comment (hunner) [inline lib/puppet/parser/functions/ip_to_cron.rb]: I don't think I understand what this is supposed to do... could you clarify?

File: lib/puppet/parser/functions/ip_to_cron.rb [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+Prototype:
+
+    ip_to_cron(n)
```
Comment (hunner) [inline lib/puppet/parser/functions/ip_to_cron.rb]: Usually the x_to_y pattern for functions means you pass an X and it returns a Y... so you pass it an IP address and it returns a cron job? Perhaps it should be called `runinterval_to_crontime()` or something?

Comment (hunner) [inline lib/puppet/parser/functions/ip_to_cron.rb]: And if you use this for many different cron jobs, they will all run at the same time... is that useful? Would it not be useful to say `command_runinterval_to_cronjob("echo my cron ran","3600")` and get back a full cron line, or ever `generate` a cron puppet resource.

I guess I just don't see the utility of this.

File: lib/puppet/parser/functions/ip_to_cron.rb (hunk 4 above)
Comment (hunner) [inline lib/puppet/parser/functions/ip_to_cron.rb]: Why is runinterval optional? Document the default, also.

File: lib/puppet/parser/functions/ip_to_cron.rb [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+The results will be as follows, varying on FQDN and network addresses:
+
+  notice: Scope(Class[main]):
+  notice: Scope(Class[main]):
+  notice: Scope(Class[main]):
+EOS
+             ) do |args|
+    ip          = lookupvar('ipaddress6') || lookupvar('ipaddress')
+    runinterval = (args[0] || 30).to_i
```
Comment (hunner) [inline lib/puppet/parser/functions/ip_to_cron.rb]: Is it possible to run a cron every 30 seconds? I thought it only fired every minute at most?

File: metadata.json [hunk 7]
```diff
@@ -1,12 +1,17 @@
 {
   "name": "puppet-extlib",
-  "version": "0.11.1-rc0",
-  "source": "https://github.com/voxpupuli/puppetcommunity-extlib",
-  "author": "puppetcommunity",
+  "version": "0.11.1-rc1",
```
Comment (hunner) [inline metadata.json]: I believe no metadata should be updated in this PR. Please submit this in a separate PR.

File: lib/puppet/parser/functions/ip_to_cron.rb (hunk 4 above)
Comment (bmjen) [inline lib/puppet/parser/functions/ip_to_cron.rb]: IMO, runinterval should be required. And the seed provided to fqdn_rand should also be a parameter too right?

File: metadata.json (hunk 7 above)
Comment (roidelapluie) [inline metadata.json]: :+1:

File: lib/puppet/parser/functions/ip_to_cron.rb (hunk 6 above)
Comment (roidelapluie) [inline lib/puppet/parser/functions/ip_to_cron.rb]: it is minutes, not seconds
